src/runtime_src/core/common/sensor.cpp

```diff
@@ -499,11 +507,13 @@ read_mechanical(const xrt_core::device * device)
   ptree_type fan_array;
   bool is_data_driven = true;
   std::vector <xq::sdm_sensor_info::data_type> output;
+  std::vector <xq::sdm_sensor_info::data_type> temp_output;
 
   //Check if requested sensor data can be retrieved in data driven model.
   try {
```

Comment:
This logic needs to be refactored.  Not clean and relies on side effects to determine if the data `is_data_driven`.

Nutshell: There should be a separate method that is called, something like `is_sensor_vmc_data() `and called by this method.

Reply:
Removed completely as we not longer need reading thermal sensors as part of mechanical report.